### PR TITLE
Fix all remaining System.Security.Cryptography.Algorithms test suite crashes.

### PR DESCRIPTION
**Pull request by jkoritzinsky**
With this + PR #48348, the test suite runs to completion and reports the following results:

> Tests run: 2527 Passed: 1285 Inconclusive: 0 Failed: 1227 Ignored: 7

So we've got a ways to go to get things passing, but now we can easily track it at least for this assembly.

### CODE REVIEW

File: src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/AesGcm.Unix.cs [hunk 1]
```diff
@@ -111,7 +111,7 @@ private void ImportKey(ReadOnlySpan<byte> key)
 
             if (plaintextBytesWritten != plaintext.Length)
             {
-                Debug.Fail($"GCM decrypt wrote {plaintextBytesWritten} of {plaintext.Length} bytes.");
+                 // Debug.Fail($"GCM decrypt wrote {plaintextBytesWritten} of {plaintext.Length} bytes.");
```
Comment (jkoritzinsky) [inline src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/AesGcm.Unix.cs:114]: I'll work on trying to isolate and fix this issue, but in the meantime this is the easiest way to keep this `Debug.Fail` method from taking down the test process.

**Review by steveisok — APPROVED** [review]

**Comment by elinor-fung** [comment]
> we can easily track it

Is this suite in some state where certain classes/groupings of tests pass now (with your other PR)?

I added `System.Security.Cryptography.Algorithms` to #45740. I was thinking we could break down the suites listed there into their test classes/collections (or whatever break-down makes sense for each suite) and mark things off as we get them working. Thoughts?

File: src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_evp_cipher.c [hunk 2]
```diff
@@ -360,12 +360,17 @@ int32_t CryptoNative_EvpCipherSetGcmTag(CipherCtx* ctx, uint8_t* tag, int32_t ta
...
 
     // Tag is provided using regular "cipher.update(tag)"
-    int32_t outl = 0;
-    uint8_t outd[1];
-    CryptoNative_EvpCipherUpdate(ctx, outd, &outl, tag, tagLength);
+    JNIEnv* env = GetJNIEnv();
+    jbyteArray inDataBytes = (*env)->NewByteArray(env, tagLength);
+    (*env)->SetByteArrayRegion(env, inDataBytes, 0, tagLength, (jbyte*)tag);
+    jbyteArray outDataBytes = (jbyteArray)(*env)->CallObjectMethod(env, ctx->cipher, g_cipherUpdateMethod, inDataBytes);
+    (*env)->DeleteLocalRef(env, outDataBytes);
```
Comment (elinor-fung) [inline src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_evp_cipher.c:373]: Delete `inDataBytes` too

File: src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_evp_cipher.c [hunk 3]
```diff
@@ -360,12 +360,17 @@ int32_t CryptoNative_EvpCipherSetGcmTag(CipherCtx* ctx, uint8_t* tag, int32_t ta
...
     assert(tagLength <= TAG_MAX_LENGTH);
 
     // Tag is provided using regular "cipher.update(tag)"
-    int32_t outl = 0;
-    uint8_t outd[1];
-    CryptoNative_EvpCipherUpdate(ctx, outd, &outl, tag, tagLength);
+    JNIEnv* env = GetJNIEnv();
+    jbyteArray inDataBytes = (*env)->NewByteArray(env, tagLength);
+    (*env)->SetByteArrayRegion(env, inDataBytes, 0, tagLength, (jbyte*)tag);
+    jbyteArray outDataBytes = (jbyteArray)(*env)->CallObjectMethod(env, ctx->cipher, g_cipherUpdateMethod, inDataBytes);
```
Comment (elinor-fung) [inline src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_evp_cipher.c:372]: `CheckJNIExceptions`?

**Review by elinor-fung — APPROVED** [review]

File: src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/AesGcm.Unix.cs (hunk 1 above)
Comment (bartonjs) [reply src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/AesGcm.Unix.cs:114]: Unlike ECB and CBC, GCM produces one byte of output for one byte of input.  Goodness isn't happening if this assert fails (and tests seemingly shouldn't be passing, unless they got the data from elsewhere having depended on this assert)

Comment (bartonjs) [reply src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/AesGcm.Unix.cs:114]: If this is needed, please don't merge without creating an issue to put it back.

Comment (jkoritzinsky) [reply src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/AesGcm.Unix.cs:114]: Oh the tests aren’t passing. But this commenter out assert allows them to run and fail and have XUnit record the issue.

Comment (jkoritzinsky) [reply src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/AesGcm.Unix.cs:114]: They fail from the exception that immediately follows this assert.

**Comment by jkoritzinsky** [comment]
There's some suites that fully pass and others that are partial. I'll try to pull together a list for the issue based on this PR.

**Comment by jkoritzinsky** [comment]
@elinor-fung I've updated the table with the list of test suites and added a link to the program I used to get the list so we can use it for the other assemblies and as we complete the test suites.

File: src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/AesGcm.Unix.cs (hunk 1 above)
Comment (jkoritzinsky) [reply src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/AesGcm.Unix.cs:114]: Tracking issue for re-enabling the assert is here: https://github.com/dotnet/runtime/issues/48471

**Pull request merged**